### ISSUE TITLE
Links change URL but not rendered component

### ISSUE THREAD

**Issue by cameronb23** [ReactTraining/react-router#4975]
I have the bulk of my code located here if you would like to look: https://gist.github.com/cameronb23/3e8841efbd09b9cb135c5702817e5563

I have my routes and navigation set up to reflect that each should direct to a certain route. However, when I click those navigation elements, it changes the URL in my browser, but does not update anything being rendered in the DOM. 

## Expected Behavior
Switch to route specified and update rendered component and URL accordingly

## Actual Behavior
Changes URL but not actual component that is supposed to be rendered on each route.

**Comment by timdorr** [comment]
This guide holds the answers you seek: https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/guides/blocked-updates.md

**Comment by FBosler** [comment]
@timdorr I am encountering a - what I think - similar problem. I looked at your guide, but I don't think it applies as I want to update URL based on the value of an input field and then rerender the component (of which the input field is part of).

See also http://stackoverflow.com/questions/43555754/react-router-v4-page-does-not-rerender-on-changed-route

Any chance you could give me a little pointer here (also if I am doing something really silly, a comment would also be appreciated)?

Thank you very much!

**Comment by alexfromapex** [comment]
I had this issue and it was because my `<Route path="/" ... />` wasn't an `exact` path so it was blocking the other Routes from matching.

**Comment by JarLowrey** [comment]
For me the issue was caused because my `Link` and my `Route` components were in different `Routers`. 

The overall App component has a Router with `Route`s. In App I added a Header with `Link`s but placed it outside `Router`. This threw an error when building the project, so I wrapped these links in another `Router`. This set up did not work.

To get it working I removed the Router from Header, moved the Header component in App into App's Router, and wrapped the Router's children in a div.

App.js
```
render(){
return (<Router >
	<div>
		<Header />
		<Switch>
			<Route exact path="/" component={Home} />
			<Route path="/externalities" component={Externalities} />
		</Switch>
	</div>
</Router>
);
}
```
